OpenRA.Mods.RA/Activities/Infiltrate.cs

```diff
@@ -56,5 +58,13 @@ protected override void OnInside(Actor self)
 				Game.Sound.PlayNotification(self.World.Map.Rules, self.Owner, "Speech",
 					notification, self.Owner.Faction.InternalName);
 		}
+
+		public override Activity Tick(Actor self)
+		{
+		if (self.Trait<Infiltrates>().IsTraitDisabled)
```

Comment:
I started writing a description but then I felt I should open a PR against your PR within an hour or two because it'd be too long to explain IMO.